CoEpi/api/CoEpiApi.swift

```diff
@@ -33,36 +33,32 @@ class CoEpiApiImpl: CoEpiApi {
     private func get<T: Decodable>(url: String) -> Single<T> {
         Single.create { emitter -> Disposable in
             let request = AF.request(url).responseJSON { response in
-                guard let data = response.data else { return }
-                do {
-                    let decoder = JSONDecoder()
-                    let obj = try decoder.decode(T.self, from: data)
-                    emitter(.success(obj))
-                } catch let error {
-                    emitter(.error(ApiError.error(
-                        message: "Couldn't parse reponse: \(error), " +
-                            "data: \(String(describing: String(data: data, encoding: .utf8)))")))
+                if let apiError = response.apiError {
+                    emitter(.error(apiError))
+                } else {
+                    guard let data = response.data else {
+                        emitter(.error(ApiError.error(message: "Response didn't have data")))
```

Comment:
Can "no response data" be a valid response?

Reply:
Wouldn't expect it for a `get` request.